server/src/main/java/org/opensearch/common/cache/stats/StatsHolder.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.stats;
+
+import org.opensearch.common.cache.ICacheKey;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.BiConsumer;
+
+/**
+ * A class caches use to internally keep track of their stats across multiple dimensions. Not intended to be exposed outside the cache.
+ */
+public class StatsHolder implements Writeable {
+
+    // The list of permitted dimensions. Should be ordered from "outermost" to "innermost", as you would like to
+    // aggregate them in an API response.
+    private final List<String> dimensionNames;
+
+    // A map from a set of cache stats dimension values -> stats for that ordered list of dimensions.
+    private final ConcurrentMap<Key, CacheStatsResponse> statsMap;
+    CacheStatsResponse totalStats;
```

Comment:
We don't need to separately track the total stats. 

If we take a snapshot of the grouped stats, we can sum all the entries to get the total stats.

Reply:
I was thinking this way would be more efficient. To sum the entries we have to iterate through all keys, and there could be many of them.